src/coord/src/catalog.rs

```diff
@@ -2227,17 +2260,30 @@ impl Catalog {
                     persist,
                 })
             }
-            Plan::CreateSource(CreateSourcePlan { source, .. }) => {
+            Plan::CreateSource(CreateSourcePlan { mut source, .. }) => {
                 let mut optimizer = Optimizer::logical_optimizer();
                 let optimized_expr = optimizer.optimize(source.expr)?;
                 let transformed_desc = RelationDesc::new(optimized_expr.typ(), source.column_names);
+                let persist_details = self.persist.source_persist_desc_from_serialized(
+                    &source.connector,
+                    source_persist_details,
+                )?;
+                // TODO: I don't like that we're injecting this into the otherwise "pristine"
```

Comment:
I don't like it either! :D But yeah def seems like what we have to do for now. I'm fine removing the TODO or leaving it and adding a bit that we'll clean it up when we start splitting SourceConnector b/c of the ingestd/dataflowd work